src/buttons/SpeedDial.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import {
+  Text,
+  View,
+  Animated,
+  StyleSheet,
+  SafeAreaView,
+  TouchableWithoutFeedback,
+} from 'react-native';
+import FAB, { FABProps } from './FAB';
+import { withTheme } from '../config';
+import { IconNode } from '../icons/Icon';
+import Color from 'color';
+import { RneFunctionComponent } from '../helpers';
+
+export type SpeedDialActionProps = Omit<FABProps, 'size'>;
+
+const SpeedDialAction: RneFunctionComponent<SpeedDialActionProps> = withTheme(
+  ({ title, titleStyle, ...actionProps }) => {
+    return (
+      <View style={styles.action}>
+        {title && <Text style={[styles.title, titleStyle]}>{title}</Text>}
+        <FAB {...actionProps} size="small" style={[actionProps.style]} />
+      </View>
+    );
+  },
+  'SpeedDialAction'
+);
+
+export type SpeedDialProps = {
+  open?: boolean;
+  openIcon?: IconNode;
+  onChange?: () => void;
```

Comment:
This seems unnecessary. Why not just `onPress`?

Reply:
onChange toggle the SpeedDial,

Reply:
Or `onClose`, `onOpen` can be used

Reply:
Yeah that's confusing with an `open` prop. Is the openness controlled by the component or by `onChange`? If there's an `onOpen` and `onClose` then having an `open` prop doesn't make sense. OR If it's just `onPress` and having the developer control an `isOpen` `prop` would also make sense. But this seems to be trying to do both at once.

Reply:
SIr SpeedDial should be closed on backdrop press,

Reply:
https://material-ui.com/api/speed-dial/#props can I do according to Material UI component?

Reply:
I suppose that should be fine.